apiserver/common/cloudspec/statehelpers.go

```diff
@@ -36,8 +41,13 @@ func MakeCloudSpecGetter(pool Pool) func(names.ModelTag) (environs.CloudSpec, er
 // any other model other than the one associated with the given
 // state.State results in an error.
 func MakeCloudSpecGetterForModel(st *state.State) func(names.ModelTag) (environs.CloudSpec, error) {
-	configGetter := stateenvirons.EnvironConfigGetter{st}
 	return func(tag names.ModelTag) (environs.CloudSpec, error) {
+		m, err := st.Model()
```

Comment:
See above comment - we can just continue to get the model as needed in the EnvironConfigGetter itself and for now just pass in state. This code here and elsewhere will be unnecessary then.